content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry/index.md

```diff
@@ -0,0 +1,247 @@
+---
+title: Collecting file-based Java logs with OpenTelemetry
+linkTitle: Collecting file-based Java logs with OpenTelemetry
+date: 
+author: >
+  [Cyrille Le Clerc](https://github.com/cyrille-leclerc) (Grafana Labs), [Gregor Zeitlinger](https://github.com/zeitlinger) (Grafana Labs)
+issue: https://github.com/open-telemetry/opentelemetry.io/issues/5606
+sig: Java, Specification
+# prettier-ignore
+cSpell:ignore: Cyrille Clerc Gregor Zeitlinger SLF4J Logback logback stdout 
+---
+Sending Java logs with OpenTelemetry typically uses the OTLP protocol. However, some scenarios require logs to be output to files or stdout due to organizational or reliability needs.
+
+A common approach to centralize logs is to use unstructured logs, parse them with regular expressions, and add contextual attributes.
+
+However, regular expression parsing is problematic. While initially easy, it becomes complex and fragile when handling all log fields, line breaks in exceptions, and unexpected log format changes. Parsing errors are inevitable with this method.
+
+## Solution
+
+The OpenTelemetry Java Instrumentation agent now offers an easy solution to convert logs from frameworks like SLF4J/Logback or Log4j2 into OTel-compliant JSON logs on stdout with all resource and log attributes.
```

Comment:
Is this unique to the java agent? I would assume that this exporter is also available for use cases where the agent is not used?

Reply:
```suggestion
The OpenTelemetry Java Instrumentation agent and SDK now offer an easy solution to convert logs from frameworks like SLF4J/Logback or Log4j2 into OTel-compliant JSON logs on stdout with all resource and log attributes.
```